sklego/meta.py

```diff
@@ -54,6 +54,47 @@ def __init__(self, estimator, groups, use_fallback=True):
         self.groups = groups
         self.use_fallback = use_fallback
 
+    def __check_group_cols_exist(self, X):
+        try:
+            x_cols = set(X.columns)
+        except AttributeError:
+            try:
+                ncols = X.shape[1]
+            except IndexError:
+                ncols = 1
+            x_cols = set(range(ncols))
+
+        # Check whether grouping columns exist
+        diff = set(as_list(self.groups)) - x_cols
+
+        if len(diff) > 0:
+            raise KeyError(f'{diff} not in columns of X ({x_cols})')
+
+    @staticmethod
+    def __check_missing_and_inf(X):
+        """Check that all elements of X are non-missing and finite"""
+        if np.any(pd.isnull(X)):
+            raise ValueError("X has NaN values")
+        try:
+            if np.any(np.isinf(X)):
+                raise ValueError("X has infinite values")
+        except TypeError:  # X cannot be converted to numeric, so checking infinites does not make sense
+            pass
+
+    def __validate(self, X, y=None):
+        try:
+            X_data = X.drop(columns=self.groups, inplace=False)
+        except AttributeError:  # np.array
+            X_data = np.delete(X, self.groups, axis=1)
+
+        if y is not None:
```

Comment:
I might add a comment here to explain that you do this because we validate during both fitting as well as prediction.